x-pack/plugins/enterprise_search/public/applications/app_search/components/engine/engine_router.tsx

```diff
@@ -73,6 +76,13 @@ export const EngineRouter: React.FC = () => {
     return clearEngine;
   }, [engineNameFromUrl]);
 
+  useEffect(() => {
+    renderHeaderActions(KibanaHeaderActions);
+    return () => {
+      renderHeaderActions(() => null);
+    };
+  }, []);
```

Comment:
Since we're defining a single future-proof `KibanaHeaderActions`, I would strongly recommend putting this in `app_search/index.tsx` > `AppSearchConfigured` and not in `EngineRouter`. Reasons why:

- We won't have to unmount / clear the header between specific routes (can lose lines 81-83)
- Technically, the header action **is** actually a completely separate React app from our app. We just happen to hydrate it with the same Kea/Redux providers that our main app has, allowing us to access state between separate React instances. Keeping it its own component that's in charge of what it conditionally renders (based on routing or whatever other state) helps us keep that conceptual distinction in a small way. Hypothetically, it also makes it easy to see all header-related logic in one place vs. jumping between views/files and trying to put together what route affects the header actions in what way.

Reply:
That's a good point. The hard part is just determining which state to key this off of. I should be able to figure it out though.

Reply:
Actually I think I can just for an `engineName` in `EngineLogic`, and if it is available, then show the button 👍

Reply:
90f49865ee72f44edde320c5e115975a18838d45

Reply:
Nice call, that works great! Another option could be using React Router's `useRouteMatch` like we do for the EngineNav:

https://github.com/elastic/kibana/blob/90f49865ee72f44edde320c5e115975a18838d45/x-pack/plugins/enterprise_search/public/applications/app_search/index.tsx#L143-L147

But engineName works great as well!